scipy/sparse/linalg/interface.py

```diff
@@ -338,6 +340,56 @@ def matmat(self, X):
 
         return Y
 
+    def rmatmat(self, X):
+        """Adjoint matrix-matrix multiplication.
+
+        Performs the operation y = A^H * x where A is an MxN linear
+        operator and x is a column vector or 1-d array, or 2-d array.
+        The default implementation defers to the adjoint.
+
+        Parameters
+        ----------
+        x : {matrix, ndarray}
+            A matrix or 2D array.
+
+        Returns
+        -------
+        y : {matrix, ndarray}
+            A matrix or 2D array depending on the type of the input.
+
+        Notes
+        -----
+        This rmatmat wraps the user-specified rmatmat routine or overridden
+        _rmatmat method to ensure that y has the correct shape and type.
+
+        """
+
+        X = np.asanyarray(X)
+
+        if X.ndim != 2:
+            raise ValueError('expected 2-d ndarray or matrix, not %d-d'
```

Comment:
In new code we shouldn't use old-style `%` string formatting.

Reply:
I am not familiar with the new-style string formatting, unfortunately. 
This is directly copy/pasted from the matmat function just above. May be we should keep the old matmat and the new rmatmat as similar as possible until someone consistently updates thew whole file?

Reply:
Yeah that's totally fine; consistency is important too.